src/components/ui/NumberFlow.tsx

```diff
@@ -41,7 +41,9 @@ export default function NumberFlow({
     value = minValue * Math.pow(10, format?.style === "percent" ? -2 : 0) * (neg ? -1 : 1);
   }
 
-  return (
+  return hideWhenZero && value == 0 ? (
```

Comment:
I find the transition during initial load to be jarring (from "-" to the value).

While the value is loading in (or undefined), let's show a loading skeleton, then once loaded show either the number flow value, or the "-".

Think minimal changes are:

Allowing value to be undefined in NumberFlow
Render skeleton if undefined
Render "-" if zero (what you have now)

Reply:
Then also replace any `val ?? 0` in all other uses of `<NumberFlow value={val ?? 0} />`

We should avoid showing 0 while loading for anything user position related as we don't want the user to think their balances are 0.